### ISSUE TITLE
Internet connection for other apps disappears when user has Adguard enabled

### ISSUE THREAD

**Issue by vbagirov** [AdguardTeam/AdguardForAndroid#284]
He uses WiFi almost exclusively, so no way to say if it happens with Mobile network. Adguard is run in VPN mode. More information and multiple logs are here:

JLC-235-67710 

Original message:
> I've been using Adguard for a few months now but lately there has been an issue on my phone that will cause Adguard to start blocking all traffic. I won't receive Hangouts messages, email notifications, Google Now won't update any cards, and I can't browse the Internet or use my connection for anything. It seems to be happening mainly on WiFi but I would say about 90% of the time I'm using my phone I'm connected to WiFi so I'm not sure if it is just isolated to a WiFi only issue. I can fix it by going into Adguard and disabling and enabling it again but that gets tedious after a while. I don't have any custom settings set up either, just the standard ones that it recommends. What might I be able to do to fix this issue for my phone?

**Comment by ameshkov** [comment]
@vbagirov could it be that he has "restricted background data" enabled?
Or some kind of "battery economy" mode?

**Comment by vbagirov** [comment]
I will check.

**Comment by evgeniy-ADG** [comment]
"I do not have anything restricted in background data. As far as battery
economy under my stock battery settings it said that Adguard is optimizing
battery use along with every other app with the exception of Google Play
Services, Project Fi, and DMService. I don't have any third party apps that
are trying to control anything with the battery or background data"

**Comment by ameshkov** [comment]
Lots of "connection timeout" messages in the log:
```
09:51:13.249 [pool-2-thread-1122] DEBUG c.a.a.filtering.commons.LogUtils - TCP id=2744 Cannot connect to /216.58.216.46:443 for com.google.android.backuptransport
java.net.SocketTimeoutException: failed to connect to /216.58.216.46 (port 443) after 5000ms
	at libcore.io.IoBridge.connectErrno(IoBridge.java:169) ~[na:0.0]
	at libcore.io.IoBridge.connect(IoBridge.java:122) ~[na:0.0]
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:183) ~[na:0.0]
	at java.net.PlainSocketImpl.connect(PlainSocketImpl.java:452) ~[na:0.0]
	at java.net.Socket.connect(Socket.java:884) ~[na:0.0]
	at java.nio.SocketChannelImpl$SocketAdapter.connect(SocketChannelImpl.java:531) ~[na:0.0]
	at com.adguard.android.filtering.vpn.VpnTcpConnection.connect(Unknown Source) ~[na:0.0]
	at com.adguard.android.filtering.vpn.VpnTcpConnection.connect(Unknown Source) ~[na:0.0]
	at com.adguard.android.filtering.vpn.VpnTcpConnection.tcpInputClosed(Unknown Source) ~[na:0.0]
	at com.adguard.android.filtering.vpn.VpnTcpConnection.addBrowserInputPacket(Unknown Source) ~[na:0.0]
	at com.adguard.android.filtering.vpn.BaseVpnConnection$1.run(Unknown Source) ~[na:0.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113) ~[na:0.0]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588) ~[na:0.0]
	at java.lang.Thread.run(Thread.java:818) ~[na:0.0]
```

**Comment by ameshkov** [comment]
@evgeniy-ADG @vbagirov 

What do I need to understand what's going on:
1. Logs with "default" logging level
2. Information when exactly (+-10 minutes) the problem occurred last time

Also it would be better if he updates to the latest release (v2.1.359).
Please ask user to provide this information.

**Comment by vbagirov** [comment]
Asked user for the default level logs.

**Comment by Bohdan-SUP** [comment]
"I updated to the most recent version and will make sure to send a report if
the problem happens again".

We'll wait for the confirmation then.

**Comment by Bohdan-SUP** [comment]
"I have not had any issues since I updated to the latest version."

**Comment by RaXorX** [comment]
Hi there, I am having the same issues with both the stable + nightly. I have tried all builds for stable and nightly which came past last month or two. I am a new user to adguard lately, been 2 months or so using it though.

My issues are the same and till now haven't had any resolution to them, support suggested some follow through things to do but they didn't solve anything at all.

I don't have the logs now but I can get from my telegram account while I was in touch with an admin from the telegram group. These are the settings I had for adguard. Funny thing was at one time I had everything disabled. Stealth mode disabled, https filtering disabled and dns filtering disabled too, there were 0 filters enabled. Adguard showed my protection level to be 1% just 1%. I enabled it at that only and let it run but after a while it still disconnected the internet for me. The only thing that brought it back was disabling and then enabling it again.

Another funny thing was that telegram x worked without any issues, I could get discord notifications but couldn't open the app. This was on oreo from where I moved to pie just in the thoughts that the rom was an issue but had the same issues on pie too. Here google play store worked too, this happened while all these apps were still being filtered by Adguard. Another thing that happened was while telegram didn't work but telegram x did. But once I got a notification for on my tg account while this issue was still there, that means the internet was cut off and telegram wouldn't show anything if not for the notification. When that notification came telegram kicked itself in from "connecting" status to suddenly having a notification. I successfully had notification in both telegram and telegram x. I confirmed that the other apps still didn't had any internet.

Hope there's a resolution to it soon or in the stable 3.0 release. For now I have to resort back to adaway and but hope to get adguard running back once again. 


[AG settings.zip](https://github.com/AdguardTeam/AdguardForAndroid/files/2937452/AG.settings.zip)
